src/appdistribution/yaml_helper.ts

```diff
@@ -0,0 +1,94 @@
+import { TestCase } from "./types";
+import * as jsYaml from "js-yaml";
+import { FirebaseError } from "../error";
+
+declare interface YamlStep {
+  goal?: string;
+  hint?: string;
+  successCriteria?: string;
+}
+
+const ALLOWED_YAML_STEP_KEYS = new Set(["goal", "hint", "successCriteria"]);
+
+declare interface YamlTestCase {
+  displayName?: string;
+  id?: string;
+  prerequisiteTestCaseId?: string;
+  steps?: YamlStep[];
+}
+
+const ALLOWED_YAML_TEST_CASE_KEYS = new Set([
+  "displayName",
+  "id",
+  "prerequisiteTestCaseId",
+  "steps",
+]);
+
+function extractIdFromResourceName(name: string): string {
+  return name.split("/").pop() ?? "";
+}
+
+function toYamlTestCases(testCases: TestCase[]): YamlTestCase[] {
+  return testCases.map((testCase) => ({
+    displayName: testCase.displayName ?? "",
+    id: extractIdFromResourceName(testCase.name!),
+    ...(testCase.prerequisiteTestCase && {
+      prerequisiteTestCaseId: extractIdFromResourceName(testCase.prerequisiteTestCase),
+    }),
```

Comment:
Can you make sure the tests cover cases where there is no prerequisiteTestCaseId, hint, and/or success criteria?